boards/riscv/it8xxx2_evb/it8xxx2_evb.dts

```diff
@@ -87,11 +83,24 @@
 	 */
 	pwm-output-frequency = <324>;
 };
+/* pwm for fan */
 &pwm7 {
 	status = "okay";
 	prescaler-cx = <PWM_PRESCALER_C4>;
 	pwm-output-frequency = <30000>;
 };
+/* fan tachometer sensor */
+&tach0 {
+	status = "okay";
+	/*
+	 * Tach0 module can select input only from [TACH0_CHANNEL_A |
+	 * TACH0_CHANNEL_B]; tach1 module can select input only from
+	 * [TACH1_CHANNEL_A | TACH1_CHANNEL_B]. We don't support that a
+	 * tach dynamically switch input from channel A/B in run time.
```

Comment:
If you added properties to the device tree to specify the TxDVS and TxCHSEL bits, then you could change the channel property to only track channel A/B.  This way it's not possible for the user to specify cross over the channel inputs.

The sample fetch routine could be simplified to
```
	if (tach_ch_valid(dev, tach_ch)) {
		/* If channel data of tachometer is valid, then save it */
		data->capture = ((*reg_fxtmrr) << 8) | (*reg_fxtlrr);
		/* Clear data valid */
		*reg_tswctlr |= config->dvs_bit;
```
Initialization would look like:
```
	if (tach_ch == TACH_CHANNEL_A) {
		/* Select TACHA output to tachometer */
		*reg_tswctlr &= ~config->chsel_bit;
		/* Clear tachometer data valid status */
		*reg_tswctlr |= config->dvs_bit;
```

The dvs_bit and chsel_bit properties would be set in the it8xxx2.dtsi file.

What do you think?

Reply:
Agree and we also needn't the tach_ch_to_pin_idx[] array.
Thanks a lot!